cmd/kaspawallet/daemon/server/sync.go

```diff
@@ -219,18 +219,30 @@ func (s *server) addEntryToUTXOSet(entry *appmessage.UTXOsByAddressesEntry, addr
 		return errors.Errorf("Got result from address %s even though it wasn't requested", entry.Address)
 	}
 
-	s.utxos[*outpoint] = &walletUTXO{
+	s.insertUTXO(&walletUTXO{
 		Outpoint:  outpoint,
 		UTXOEntry: utxoEntry,
 		address:   address,
-	}
+	})
 
 	return nil
 }
 
+// insertUTXO inserts the given utxo into s.utxosSortedByAmount, while keeping it sorted.
+func (s *server) insertUTXO(utxo *walletUTXO) {
+	s.utxosSortedByAmount = append(s.utxosSortedByAmount, utxo)
+	// bubble up the new UTXO to keep the UTXOs sorted by value
+	index := len(s.utxosSortedByAmount) - 1
```

Comment:
This can be a binary search

Reply:
e.g. same logic as here https://github.com/kaspanet/kaspad/blob/master/domain/consensus/processes/reachabilitymanager/ordered_tree_node_set.go#L42

Reply:
You can also use `sort.Search`, like in `findTransactionIndex`

Reply:
nit: I think it's more readable to use code such as `func (tobf *TransactionsOrderedByFeeRate) Push`, and maybe even more efficient